### DISCUSSION TITLE
Ignoring .DS_Store on upload

### DISCUSSION THREAD

**Discussion by alexmf23** [pimcore/pimcore#12568] [Q&A / Support]
Hello,
on Mac when trying to upload a folder into assets and there's a .DS_Store file inside it throws an error.
Is it possible to ignore .DS_Store without errors to pop-up ?
<img width="633" alt="Screenshot 2022-07-01 at 14 24 29" src="https://user-images.githubusercontent.com/63473281/176895760-e3bc9fb6-8c28-404d-af60-69eb0b7a4e2e.png">
<img width="864" alt="Screenshot 2022-07-01 at 14 24 37" src="https://user-images.githubusercontent.com/63473281/176895767-b1047275-f03e-4a49-a849-cde066624063.png">

**Comment by skynet-technologies** [comment]
`.DS_Store` is a hidden file (because the .) and you can‘t just see it. In the terminal, you can see it with `ls -la`.

 

Then, if you have already once pushed `.DS_Store` to your repo, you must untrack this file, then only untracked files will be ignored with `.gitignore`.

 

To untrack a file, execute `git rm --cached .DS_Store`. Then, when `.DS_Store` is correctly registered in `.gitignore`, you will never see this file on your remote repository.

 

But there is also no problem, to overwrite the `.DS_Store`-file with a merge. This file contains information about system configuration. And for every person where something push to this repository from a Mac OS, will create their own `.DS_Store`. But with correct adding to `.gitignore`, this file is no longer overwritten every time.

You can also remove the file and try again. Please refer to this document to delete it.
https://support.microsoft.com/en-us/office/remove-ds-store-files-on-macos-d2f8dca0-740a-4f7e-89b9-5b2cbbc50386

Hope it helps.

**Reply by alexmf23** [reply]
Yes, but this is not what was meaning.
Let's say u have a production app, and from a mac u drag a folder into the assets to upload. Usually that folder has a .DS_Store inside so this message pops up and it's annoying for someone who doesnt know. The file uploader should automatically ignore this file.

**Reply by skynet-technologies** [reply]
You can extend the Asset model in the section and can allow `.DS_Store` extension.
Currently, by default Pimcore ignores them.
Learn more here:
https://github.com/pimcore/pimcore/blob/10.x/models/Asset.php#L1878

**Reply by jdreesen** [reply]
@skynet-technologies the line you linked to does not exist ;)

**Reply by skynet-technologies** [reply]
Hey, @jdreesen, it's in the `getTypeFromMimeMapping` method on line no 434
Need to customize that method.
https://github.com/pimcore/pimcore/blob/10.x/models/Asset.php#L434

**Reply by cancan101** [reply]
Is there anyway to tell Pimcore to ignore these files when someone dropped a folder / zip into Pimcore? Adding them as assets is not useful but our users do this from time to time.

**Reply by BlackbitDevs** [reply]
Have created PR https://github.com/pimcore/admin-ui-classic-bundle/pull/832
